src/pages/leaderboard.tsx

```diff
@@ -1,70 +1,81 @@
-import { useCallback, useEffect, useRef, useState } from "react";
+import { useEffect, useMemo } from "react";
 import { useSession } from "next-auth/react";
-import { Box, Typography } from "@mui/material";
-import { type LeaderboardData } from "~/types/index.js";
+import { CircularProgress, Typography } from "@mui/material";
+
 import { isValidSession } from "~/utils/session.js";
 import { AddScoreTable, LeaderboardTable } from "~/components/tables/index.js";
-import { leaderboardData } from "~/constants/index.js";
 import { text } from "~/assets/text/index.js";
 import { styles } from "~/styles/pages/leaderboard.js";
-import { fakeCountdown } from "~/constants/fake-data/landing";
+import { fakeCountdown } from "~/constants/fake-data/landing.js";
+import { api } from "~/utils/api.js";
+import { isClient } from "~/utils/client.js";
+
+const USERS_PER_PAGE = 20;
 
 const LeaderBoard = () => {
-  const [currentPage, setCurrentPage] = useState<number>(1);
-  const [currentData, setCurrentData] = useState<LeaderboardData[]>([]);
-  const itemsPerPage = 6;
   const { data: sessionData } = useSession();
   const isAuthenticated = isValidSession(sessionData);
+
   const isGamePlayed = true;
 
-  const loadMoreData = useCallback(() => {
-    const startIndex: number = (currentPage - 1) * itemsPerPage;
-    const endIndex: number = startIndex + itemsPerPage;
-    const newData: LeaderboardData[] = leaderboardData.slice(0, endIndex);
+  const getRankedUsers = api.user.getRankedUsers.useInfiniteQuery(
```

Comment:
Shouldn't this be imported from db or some other service?

Reply:
I doubt it would be a good idea to call the DB from the client